platform-sdk/consensus-otter-tests/src/testFixtures/java/org/hiero/otter/fixtures/metric/NodeMetrics.java

```diff
@@ -0,0 +1,233 @@
+// SPDX-License-Identifier: Apache-2.0
+package org.hiero.otter.fixtures.metric;
+
+// SPDX-License-Identifier: Apache-2.0
+
+import com.swirlds.metrics.api.Metric;
+import com.swirlds.metrics.api.MetricConfig;
+import com.swirlds.metrics.api.Metrics;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Proxy;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.stream.Collectors;
+import org.hiero.consensus.model.node.NodeId;
+
+/**
+ * In-memory implementation of the {@link Metrics} interface that is tailored for usage in otter
+ * integration tests. It records the full history of every metric value so that assertions can be
+ * executed after a test run.
+ */
+public class NodeMetrics implements Metrics {
+
+    /** The node this metrics instance belongs to */
+    private final NodeId nodeId;
+
+    /** Map<identifier, metric> */
+    private final Map<String, Metric> metrics = new ConcurrentHashMap<>();
+
+    /** Map<identifier, history> */
+    private final Map<String, List<Object>> history = new ConcurrentHashMap<>();
+
+    /** Updaters that should be executed – executed on {@link #start()} in the test JVM */
+    private final Set<Runnable> updaters = new CopyOnWriteArraySet<>();
+
+    /** Flag indicating whether start() has been invoked */
+    private volatile boolean started = false;
+
+    public NodeMetrics(@NonNull final NodeId nodeId) {
+        this.nodeId = Objects.requireNonNull(nodeId, "nodeId must not be null");
+        // auto-register to the central singleton
+        NetworkMetrics.getInstance().register(this);
+    }
+
+    /**
+     * Returns the node id.
+     *
+     * @return the id
+     */
+    @NonNull
+    public NodeId getNodeId() {
+        return nodeId;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void start() {
+        if (started) {
+            return;
+        }
+        started = true;
+        final Thread updaterThread = new Thread(
+                () -> {
+                    while (started) {
+                        final Instant start = Instant.now();
```

Comment:
We should not use `Instant.now()` but use the `FakeTime` instead.

Reply:
On second thought. If this implementation is only used in Turtle tests, we should make `NodeMetrics` a `TimeTickReceiver` and call the updaters every time a tick is received. What do you think?